packages/frontend/src/components/Board/Cell.tsx

```diff
@@ -73,20 +73,30 @@ export const Cell: FC<CellProps> = memo((props) => {
     <div
       key={id}
       className={`border-l w-full aspect-square relative overflow-hidden ${
-        gameObject && (gameObject.type as GameObjectType)
+        gameObject && gameObject.type
           ? 'flex items-center justify-center'
           : 'justify-center'
       }`}
     >
-      {gameObject && (gameObject.type as GameObjectType) && (
+      {gameObject && gameObject.type && (
         <div className="flex flex-col w-full">
           <p className="text-[6px] text-black dark:text-white max-w-[98%] self-center sm:text-[10px] overflow-hidden whitespace-nowrap truncate">
             {getCharacterName(gameObject)}
           </p>
           <img
             src={getGameCharacter(gameObject)}
-            className="w-[70%] h-[70%] self-center"
+            className={`w-[70%] h-[70%] self-center ${
+              gameObject.type.includes('Teleport') ? 'rotate' : ''
+            } ${gameObject.type.includes('DiamondGame') ? 'diamond' : ''}`}
```

Comment:
Suggestion: instead of letting the cell do all these checks to decide how things should look we should create new components for each game object where they themselves can decide how they want to look

Reply:
maybe it should be another PR though.

Reply:
So the cell component should decide the size of the cell but the content should be delegated to a game object component for the specific type of game object

Reply:
Created new pr #324 to break out game object component from cell. Includes the earlier commits from branch animation.